src/components/AdminRegisterPage/index.jsx

```diff
@@ -1,46 +1,51 @@
-import React, { useMemo } from 'react';
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { Redirect } from 'react-router-dom';
-import {
-  Container, Row, Col, Alert, MailtoLink,
-} from '@edx/paragon';
+import { Container, Row, Col } from '@edx/paragon';
 
 import LoadingMessage from '../LoadingMessage';
 
-import { redirectToProxyLogin } from '../../utils';
+import {
+  getProxyLoginUrl,
+  redirectToProxyLogin,
+  hasEnterpriseAdminRole,
+} from '../../utils';
+import apiClient from '../../data/apiClient';
 
 const AdminRegisterPage = ({ authentication, match }) => {
-  const enterpriseSlug = useMemo(
-    () => match.params.enterpriseSlug,
-    [match.params],
+  const { enterpriseSlug } = match.params;
+
+  useEffect(
+    () => {
+      if (!authentication.username) {
+        // user is not authenticated
+        return;
+      }
+
+      if (!hasEnterpriseAdminRole(authentication.roles)) {
+        // user is authenticated but doesn't have the `enterprise_admin` JWT role; force a log out so their
+        // JWT roles gets refreshed. on their next login, the JWT roles will be updated.
```

Comment:
i'm curious if there is a way to refresh the JWT without forcing the user to log out and log back in? like just have the frontend re-request the JWT? (doesn't need to be addressed here if that's too much/not possible)

Reply:
Unfortunately, there's no easy way given our use of the legacy edx/frontend-auth package to manage who the authenticated user is.

The `authentication` variable above comes from the Redux store and edx/frontend-auth doesn't expose a way to update/modify what is stored there. 

I tried to utilize the `login_refresh` endpoint to avoid a forced log out but without the user going through the logistration flow again (and triggering the appropriate Django signal), the user won't have any additional roles that may have been added after the user initially logged in. Even if that did work, there's no way given edx/frontend-auth to update the `authentication` value stored in Redux at the moment.